src/shared/condition.c

```diff
@@ -1134,7 +1137,8 @@ static int condition_test_psi(Condition *c, char **env) {
                 return log_debug_errno(r, "Failed to parse loadavg: %s", c->parameter);
 
         r = read_resource_pressure(pressure_path, PRESSURE_TYPE_FULL, &pressure);
```

Comment:
we shouldn't even ask for this if we are looking at the per-system file. i.e. this call should already be conditionalized.

maybe instead of a "bool system_wide" just keep a "bool PressureType preferred_pressure_type". that one you either initialize to PRESSUE_TYPE_FULL or to PRESSURE_TYPE_SOME int he big if block above.

Then you just pass that here as 2nd arg, so that we ask for the right thing right away.

And then you conditionalize the 2nd call to read_resource_pressure() below not on the the system_wide bool but instead simply whether preferred_pressure_type is PRESSURE_TYPE_FULL currently.

Reply:
Thanks for the guidance, 

PressureType preferred_pressure_type is now set at the beginning to 'full'.

If the check is at system level and for cpu pressure then preferred_pressure_type is set to 'some'.

And the error handling for -ENODATA is simplified again because it should only ever occur if a kernel without cpu.pressure 'full' support for cgroups was used.